File: Codigos estudiantes por lenguaje/PY/Bryann Valderrama/BitWise/maxSubArrayXOR.py
from sys import stdin, stdout, maxsize
rl = stdin.readline
wr = stdout.write


def maxSubArrayXOR(arr, n):
    ans = -maxsize - 1
    for i in range(n):
        curr_xor = 0
        for j in range(i, n):
            curr_xor = curr_xor ^ arr[j]
            ans = max(ans, curr_xor)
    return ans


arr = [8, 1, 2, 12]
n = len(arr)
wr(f'Maximo SubArray XOR es {maxSubArrayXOR(arr,n)}')

File: Codigos estudiantes por lenguaje/Java/Bryann Valderrama/BitWise/maxSubArrayXOR.java
public class Main {

    static int maxSubarrayXOR(int arr[], int n) {
        int ans = Integer.MIN_VALUE; // Inicializar resultado 
        // Escogiendo puntos de inicio para los subarreglos
        for (int i = 0; i < n; i++) {
            // para guardar XOR del actual subarreglo
            int curr_xor = 0;
            // Escogiendo puntos finales de subArreglos empezando por i
            for (int j = i; j < n; j++) {
                curr_xor = curr_xor ^ arr[j];
                ans = Math.max(ans, curr_xor);
            }
        }
        return ans;
    }

    public static void main(String args[]) {
        int arr[] = {8, 1, 2, 12, 0, 3, 4, 5, 12, 1221};
        int n = arr.length;
        System.out.println("Maximo subarray XOR es " + maxSubarrayXOR(arr, n));
    }
}
